### PR TITLE
layer.conf: remove '-mbranch-protection=standard' globally on aarch64

### PR DESCRIPTION
**Pull request by quaresmajose**
On aarch64 the handling of runtime exception is broken when build with clang.
There is currently a fix for libcxx on the layer but it is not enough and this same fix needs to be applied globally.

First fix attempt just include libcxx:
https://github.com/kraj/meta-clang/issues/963
https://github.com/kraj/meta-clang/pull/968

Remaining reported issue:
https://github.com/kraj/meta-clang/issues/998

---
### Contributor checklist

- [x] Changes have been tested
- [x] `Signed-off-by` is present
- [x] The PR complies with the [Open Embedded Commit Patch Message Guidelines](http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)

### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

### CODE REVIEW

File: conf/layer.conf [hunk 1]
```diff
@@ -35,5 +35,8 @@ CLANGSDK ??= "0"
 
 LLVMVERSION = "18.1.8"
 
+# introduced in oe-core 8905639d breaks the handling of runtime exception when build with clang
+TUNE_CCARGS:remove:toolchain-clang = "${@bb.utils.contains('TUNE_FEATURES', 'aarch64', '-mbranch-protection=standard', '', d)}"
+
```
Comment (kraj) [inline conf/layer.conf:40]: lets move this to `nonclangable.conf`

**Review by kraj — CHANGES_REQUESTED** [review]

File: conf/layer.conf (hunk 1 above)
Comment (kraj) [reply conf/layer.conf:40]: and check if https://github.com/kraj/meta-clang/pull/999 made any difference

Comment (quaresmajose) [reply conf/layer.conf]: I have tested it and unfortunately don't solve the issue, also reported that on https://github.com/kraj/meta-clang/issues/998

**Review by kraj — APPROVED** [review]
LGTM

**Pull request merged**